documentation/configuration.markdown

```diff
@@ -9,9 +9,7 @@ canonical: "/puppetserver/latest/configuration.html"
 [`puppetserver.conf`]: ./config_file_puppetserver.markdown
 [deprecated]: ./deprecated_features.markdown
 
-Puppet Server honors most settings in `puppet.conf` and picks them up automatically. However, for some tasks, such as configuring the web server or an external Certificate Authority (CA), Puppet Server has separate configuration files and settings.
-
-These files and settings are described below. For more information about differences between Puppet Server and the Ruby Puppet master's use of `puppet.conf` settings, see  [Puppet Server: Differing Behavior in `puppet.conf`](./puppet_conf_setting_diffs.markdown).
+Puppet Server automatically loads settings in the `main` and `master` section, and honors most other settings in [`puppet.conf`](https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/services/config/puppet_server_config_core.clj#L18-L49). However, for some tasks, such as configuring the web server or an external Certificate Authority (CA), Puppet Server has separate configuration files and settings. These files and settings are described below. For more information about differences between Puppet Server and the Ruby Puppet master's use of `puppet.conf` settings, see  [Puppet Server: Differing Behavior in `puppet.conf`](./puppet_conf_setting_diffs.markdown).
```

Comment:
Puppet Server will prefer values set in the master section to the same settings set in the main section, if there are duplicates.

eg, the config file:
```
[master]
setting1 = foo
setting2 = baz

[main]
setting1 = bar
setting3 = buzz
```

Will resolve `setting1` to `foo`, `setting2` to `baz`, `setting3` to `buzz`, and `setting4` will be whatever its default value is.

Puppet Server will only do the above (load and honor) those values listed in the linked document.

fwiw, I think it'd be better to write out those settings in the docs than link to the code. In code they start with colons and  use hyphens, however when writing the actual settings the colons are dropped and the hyphens are turned into underscores, eg the linked `:allow-duplicate-certs` becomes the setting `allow_duplicate_certs`